### PR TITLE
Cleanup the test directory

### PR DESCRIPTION
**Pull request by stefwalter**
This directory has grown significantly, with the integration of
the avocado tests, and now the container test suite .

Lets clean it up:

 * common: Common stuff applicable to most test suites.
 * images/scripts: Scripts for creating images.
 * verify/files: Files for the verify test suite.
 * verify/naughty: Known issues for the verify test suite.

No doubt there will be further work to put stuff in place that has been missed, and this is an initial round of cleanup.

### CODE REVIEW

**Comment by petervo** [comment]
files is used by tests that aren't in verify. Should that be top level? Or in common?

**Comment by stefwalter** [comment]
> files is used by tests that aren't in verify. Should that be top level? Or in common?

Because they share code with verify right? In that case I thought thaht the canonical use of those files test suite was in ./verify/ and others borrowed them. Is that correct?

**Comment by petervo** [comment]
There is shared code which means shared files but also but this file is only used from image tests: https://github.com/cockpit-project/cockpit/blob/master/test/files/cockpit-openshift-template.json.

We could have every test type keep it's own set of files. Or have files top-level/common

**Comment by stefwalter** [comment]
> There is shared code which means shared files but also but this file is only used from image tests: 

Moved into containers tests.

**Comment by petervo** [comment]
Not containers/files/...? In the short term there will be at least one more of these for the registry.

**Comment by stefwalter** [comment]
> Not containers/files/...? In the short term there will be at least one more of these for the registry.

Yup that's fine too.

**Comment by stefwalter** [comment]
Blocked on #3850

**Comment by dperpeet** [comment]
needs rebase

**Comment by stefwalter** [comment]
Rebased, lets see how this goes.

**Comment by stefwalter** [comment]
Rebased, lets see if this does the trick.

File: test/avocado/run-tests [hunk 1]
```diff
@@ -30,12 +30,12 @@ import glob
 # the directory structure breaks on the second run with .pyc files
 sys.dont_write_bytecode = True
 
-import testinfra
-import testvm
-
-topdir = os.path.normpath(os.path.join(os.path.dirname(os.path.realpath(__file__)), ".."))
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
```
Comment (dperpeet) [inline test/avocado/run-tests]: Shouldn't we either use realpath for both of these lines or for none?
realpath seems like the safer option

File: test/avocado/run-tests [hunk 2]
```diff
@@ -52,7 +52,8 @@ def prepare_avocado_tests(machine):
     """ Upload all avocado related files and create the results directory
     """
     machine.execute(command="mkdir -p " + avocado_results_dir)
-    machine.upload([os.path.join(machine.test_dir, "avocado")], machine_test_dir)
+    machine.upload(["avocado"], machine_test_dir)
+    machine.execute(command="ls -l /tmp/avocado_tests")
```
Comment (dperpeet) [inline test/avocado/run-tests]: This should use the variable `machine_test_dir` instead of hardcoding the path.

File: test/common/testlib.py [hunk 3]
```diff
@@ -682,10 +680,11 @@ def start(self):
         environ = os.environ.copy()
         if self.lang:
             environ["LC_ALL"] = self.lang
+        path = os.path.dirname(__file__)
```
Comment (dperpeet) [inline test/common/testlib.py]: this got rid of the realpath... is that on purpose?

File: test/verify/parent.py [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import sys
 import os
 
-sys.path.append(os.path.dirname(os.path.dirname(__file__)))
+sys.path.append(os.path.join(os.path.dirname(os.path.dirname(__file__))))
```
Comment (dperpeet) [inline test/verify/parent.py]: is the double dirname for a ..?

File: test/clean-testdir [hunk 5]
```diff
@@ -1,117 +0,0 @@
-#!/bin/bash
```
Comment (dperpeet) [inline test/clean-testdir]: This file gets removed completely? Why?

**Comment by dperpeet** [comment]
debian tests fails with "not found"

**Comment by dperpeet** [comment]
also needs rebase

**Comment by stefwalter** [comment]
> The link on github to the logs doesn't work - I think because log.html was moved. Should we update the test runners first to support both locations?

The runners have moved. It seems like a bug, outside of this cleanup, that it's not picking up the log.html stuff in a different location. I could spend all day fixing all the bugs in all of this (including debian-unstable's ssh support), but at some point I need to do real work. Not sure what to do.

File: test/avocado/run-tests (hunk 1 above)
Comment (stefwalter) [inline test/avocado/run-tests]: Why is it safer?

File: test/avocado/run-tests (hunk 2 above)
Comment (stefwalter) [inline test/avocado/run-tests]: Oh, this line should be removed. Will do.

File: test/clean-testdir (hunk 5 above)
Comment (stefwalter) [inline test/clean-testdir]: It should just be part of 'make clean'

File: test/verify/parent.py (hunk 4 above)
Comment (stefwalter) [inline test/verify/parent.py]: Yup.

File: test/common/testlib.py (hunk 3 above)
Comment (stefwalter) [inline test/common/testlib.py]: Yes, we don't need realpath here.

File: test/avocado/run-tests (hunk 1 above)
Comment (dperpeet) [inline test/avocado/run-tests]: This was an issue when we had symlinks to files. If the avocado tests pass without the realpath, then I think we should leave it out.

Comment (dperpeet) [inline test/avocado/run-tests]: To make myself clear: I don't think we necessarily need to change this now if it works.

**Comment by dperpeet** [comment]
needs rebase

**Comment by dperpeet** [comment]
the log.html issues needs fixing before this gets merged, I think

**Comment by stefwalter** [comment]
> the log.html issues needs fixing before this gets merged, I think

@mvollmer Do you know why the log.html stuff doesn't work when we move the URL? I have no explanation. When I run it manually it works fine.

**Comment by stefwalter** [comment]
Okay, log.html should be fixed now. Was caused because of the fact that github-task is running master, not the branch. Changed the code to reflect this reality.

**Comment by stefwalter** [comment]
Blocked on #3998

**Comment by stefwalter** [comment]
Fixed two more file path issues.

**Comment by stefwalter** [comment]
This rebase happens without conflict. In addition the test failure above is related to an upstream polkit bug that has been fixed elsewhere.

**Comment by stefwalter** [comment]
Rebased and  testing again, since hopefully that'll help this get in faster.

**Pull request closed without merging**